src/SmartComponents/DriftPage/DriftTable/DriftTable.js

```diff
@@ -17,30 +17,46 @@ import AddSystemButton from '../AddSystemButton/AddSystemButton';
 class DriftTable extends Component {
     constructor(props) {
         super(props);
-        this.systemIds = queryString.parse(this.props.location.search).system_ids;
+        this.setSystemIds();
         this.fetchCompare = this.fetchCompare.bind(this);
+        this.removeSystem = this.removeSystem.bind(this);
         this.formatDate = this.formatDate.bind(this);
     }
 
     async componentDidMount() {
         await window.insights.chrome.auth.getUser();
-        if (this.systemIds) {
-            this.fetchCompare(this.systemIds);
-        }
+        this.fetchCompare(this.systemIds);
+    }
+
+    setSystemIds() {
+        this.systemIds = queryString.parse(this.props.location.search).system_ids;
+        this.systemIds = Array.isArray(this.systemIds) ? this.systemIds : [ this.systemIds ];
+        this.systemIds = this.systemIds.filter(item => item !== undefined);
     }
 
     formatDate(dateString) {
         let date = new Date(dateString);
         return date.toLocaleDateString() + ' ' + date.toLocaleTimeString();
     }
 
+    removeSystem(systemId) {
+        let ids = this.systemIds.filter(item => item !== systemId);
```

Comment:
If you:
- load the blank state
- choose 2+ systems
- immediately click one of the Xs

You lose all of the systems you selected. systemIds at this point is empty, so it runs a compare on nothing. Refreshing the page with the system Ids in the url will load them into systemIds.

Reply:
I'm having a hard time re-producing this issue. The loading screen covers up the 'x' to click it for me preventing any issues?

Reply:
I think this was cleared up with your fix.